engine.py

```diff
@@ -39,6 +39,18 @@ def init_engine(self):
         else:
             self._logger.setLevel(logging.INFO)
 
+        # Figure out which implementation we will use.  If the tk instance
```

Comment:
Need to do this check beforehand so that _define_qt_base, which can't be specialized inside the engine's implementation class, can add a workaround for a Windows dialog visibility bug.